### ISSUE TITLE
HTTP Error: 400, database.rulesString is not of a type(s) string when deploying

### ISSUE THREAD

**Issue by yotamelkaslasy** [r-park/todo-react-redux#51]
Hello,
I wanted to simulate a deploy with new firebase app. I followed all the instructions and when I deploy there's an issue with the rules I assume.

Here's the console error:

>$ firebase deploy

> === Deploying to 'todo-react-redux'...
> 
> i  deploying database, hosting
> ✔  database: rules ready to deploy.
> i  hosting: preparing target directory for upload...
> Uploading: [==================================      ] 86%✔  hosting: target folder uploaded successfully
> ✔  hosting: 7 files uploaded successfully
> i  starting release process (may take several minutes)...
> 
> Error: HTTP Error: 400, database.rulesString is not of a type(s) string

I'm new with firebase, any ideas ?

I should mention, when I tried to 'npm run build',  karma couldn't open chrome and it canceled the process. after I excluded the test from the build, the app built fine.

**Comment by r-park** [comment]
Hi @yotamelkaslasy — it's difficult to diagnose without some code examples. Did you make any changes/customizations?

I tried creating a new firebase and was able to deploy this repo. The only changes I made were to update the project [config](https://github.com/r-park/todo-react-redux/blob/master/src/core/firebase/config.js) and [name](https://github.com/r-park/todo-react-redux/blob/master/.firebaserc):
```shell
$ firebase use default
Now using alias default (test-issue-51)
$ firebase deploy

=== Deploying to 'test-issue-51'...

i  deploying database, hosting
✔  database: rules ready to deploy.
i  hosting: preparing target directory for upload...
✔  hosting: 7 files uploaded successfully
i  starting release process (may take several minutes)...

✔  Deploy complete!

Hosting Site: https://test-issue-51.firebaseapp.com
Dashboard: https://console.firebase.google.com/project/test-issue-51/overview

```

**Comment by asndev** [comment]
hi,

im getting the same error with a fresh copy of the repository (just changed project name and config file).

**Comment by andrerpena** [comment]
I'm also getting this error... 

what I changed (all to reflect my project configuration):

- .firebaseerc
- /src/core/firebase/config.js 

I just followed the instructions.. Any idea?

**Comment by yotamelkaslasy** [comment]
Thanks @r-park 

I managed to deploy with :
`sudo firebase deploy --only hosting` 

Its a bug on firebase end.

I'm using firebase 3.0.6 (latest currently)

If you run into meaningless firebase errors try to debug with :
`sudo firebase deploy --debug`
and see where the process fails.

**Comment by r-park** [comment]
Thanks for the update @yotamelkaslasy 👍 

@asndev and @andrerpena — can you please try @yotamelkaslasy's solution?

**Comment by asndev** [comment]
`--only hosting` is working for me, cheers for the effort.

**Comment by r-park** [comment]
I updated to `firebase-tools 3.0.6` and was able to reproduce the error. However deploying with:
```shell
firebase deploy --only hosting
``` 
will not include the database rules contained in `firebase.json`. 

The error seems to arise because `firebase-tools 3.0.6` is expecting the `database` field in `firebase.json` to reference an external file. I've moved the rules to `firebase.rules.json` which fixed the error.
